src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/GetUnallocatedCaseServiceTest.kt

```diff
@@ -69,7 +73,36 @@ internal class GetUnallocatedCaseServiceTest {
     coEvery { mockWorkforceAllocationsToDeliusApiClientClient.getDeliusCaseDetailsCases(emptyList()) } returns emptyFlow()
 
     val cases =
-      GetUnallocatedCaseService(mockRepo, mockOutOfAreaTransferService, mockk(), mockWorkforceAllocationsToDeliusApiClientClient).getAllByTeam("TM1")
+      GetUnallocatedCaseService(mockRepo, mockOutOfAreaTransferService, mockk(), mockWorkforceAllocationsToDeliusApiClientClient, mockLaoService).getAllByTeam("TM1")
+        .toList()
+
+    verify(exactly = 0) { mockWorkforceAllocationsToDeliusApiClientClient.getDeliusCaseDetailsCases(listOf(unallocatedCaseEntity)) }
+    assertEquals(0, cases.size)
+  }
+
+  @Test
+  fun `must not return unallocated cases which are restricted `() = runBlocking {
+    val crn = "X123456"
+    val unallocatedCaseEntity = UnallocatedCaseEntity(
+      name = "restricted",
+      crn = crn,
+      providerCode = "PC1",
+      teamCode = "TM1",
+      tier = "C2",
+      id = 2L,
+      convictionNumber = 1,
+    )
+
+    every { mockRepo.findByTeamCode("TM1") } returns listOf(unallocatedCaseEntity)
+    coEvery { mockWorkforceAllocationsToDeliusApiClientClient.getUserAccess(listOf(crn)) } returns
+      DeliusUserAccess(
+        access = listOf(DeliusCaseAccess(crn, userRestricted = true, false)),
```

Comment:
Suggesting either adding parameter names for all or none

Reply:
added all